src/promptflow-evals/promptflow/evals/evaluators/election/_eci.py

```diff
@@ -0,0 +1,89 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from promptflow._utils.async_utils import async_run_allowing_running_loop
+from promptflow.evals._common.constants import _InternalEvaluationMetrics
+from promptflow.evals._common.rai_service import evaluate_with_rai_service
+
+
+class _AsyncECIEvaluator:
+    def __init__(self, project_scope: dict, credential=None):
+        self._project_scope = project_scope
+        self._credential = credential
+
+    async def __call__(self, *, question: str, answer: str, **kwargs):
+        # Validate inputs
+        # Raises value error if failed, so execution alone signifies success.
+        if not (question and question.strip() and question != "None") or not (
+            answer and answer.strip() and answer != "None"
+        ):
+            raise ValueError("Both 'question' and 'answer' must be non-empty strings.")
+
+        # Run score computation based on supplied metric.
+        result = await evaluate_with_rai_service(
+            metric_name=_InternalEvaluationMetrics.ECI,
+            question=question,
+            answer=answer,
+            project_scope=self._project_scope,
+            credential=self._credential,
+        )
+        return result
+
+
+class ECIEvaluator:
```

Comment:
I am not sure if having "election" as the folder name here is okay. Why not just use "_eci"?

Reply:
Also, should we add underscore prefix to the class name to make it looks "internal"?

Reply:
good point on the package name, as for the class name, I don't think that's needed, since the package itself will be internal, and Minsoo mentioned that she'd rather avoiding designing the classes to need breaking changes if we ever went GA with this.